### ISSUE TITLE
🔥 influxd "context canceled" error, queries never complete 🔥

### ISSUE THREAD

**Issue by max0x7ba** [influxdata/influxdb#23907]
I have a Python script which makes 66 read queries into a local InfluxDB database using 32 clients. It measures response time of each query and the total time. With version 2.0.4 the total time for these queries to complete was 50 seconds. 

These read queries only differ in measurement filter:
```
from(bucket:"A")
 |> range(start: 1670617080, stop: 1664578800)
 |> filter(fn: (r) => r._measurement == "M" and (r._field == "F1" or r._field == "F2" or r._field == "F3" or r._field == "F4" or r._field == "F5" or r._field == "F6"))
 |> pivot(rowKey: ["_time"], columnKey: ["_field"], valueColumn: "_value")
 |> keep(columns: ["_time","F1","F2","F3","F4","F5","F6"])
 |> map(fn: (r) => ({r with _time: uint(v: r._time)}))
```

and they return tables with 1 `uint` and 6 `float` columns and up to ~100,000 rows.

I upgraded from version 2.0.4 to 2.5.1. Now only 34 out of 66 queries complete while the rest get stuck forever. When that happens `influxd` writes the following error message (~20 times a minute) into the system log:

```
msg="internal error not returned to client" log_id=0e7eiZSl000 handler=error_logger error="context canceled"
```

I downgraded through earlier versions until I found one that completes all the queries - version 2.0.7. With version 2.0.8 half of the queries never complete.

With version 2.0.4 the total time for these queries was 50 seconds. With version 2.0.5 the total query time doubled.

__Expected behavior:__
66 read queries complete.

__Actual behavior:__
34 out of 66 read queries complete while the rest get stuck forever.

__Environment info:__

* System info: `Linux 5.15.0-52-lowlatency x86_64`, Ubuntu 20.04 LTS.
* InfluxDB version: `InfluxDB v2.5.1 (git: 5b6fdbf05d) build_date: 2022-11-02T18:06:28Z`
* Other relevant environment details: `influxdb-client 1.34.0`.

__Config:__
```
query-concurrency = 32
query-queue-size = 32
```

__Logs:__
```
Nov 13 01:50:56 influxd-systemd-start.sh[3874]: ts=2022-11-13T01:50:56.555889Z lvl=warn msg="internal error not returned to client" log_id=0e7eiZSl000 handler=error_logger error="context canceled"
Nov 13 01:51:00 influxd-systemd-start.sh[3874]: ts=2022-11-13T01:51:00.051461Z lvl=warn msg="internal error not returned to client" log_id=0e7eiZSl000 handler=error_logger error="context canceled"
Nov 13 01:51:02 influxd-systemd-start.sh[3874]: ts=2022-11-13T01:51:02.330052Z lvl=warn msg="internal error not returned to client" log_id=0e7eiZSl000 handler=error_logger error="context canceled"
Nov 13 01:51:03 influxd-systemd-start.sh[3874]: ts=2022-11-13T01:51:03.608479Z lvl=warn msg="internal error not returned to client" log_id=0e7eiZSl000 handler=error_logger error="context canceled"
```

__Performance:__
[iostat.txt](https://github.com/influxdata/influxdb/files/9996356/iostat.txt)
[profiles.tar.gz](https://github.com/influxdata/influxdb/files/9996357/profiles.tar.gz)

**Comment by kazomotos** [comment]
+1

**Comment by Jibec** [comment]
same issue here, using holtWinters on a large dataset
it were working fine a few releases ago and now fail if there is more than a few groups

**Comment by Jibec** [comment]
[2022-12-08_13 45_influxdb_data.csv](https://github.com/influxdata/influxdb/files/10185528/2022-12-08_13.45_influxdb_data.csv)
here is my dataset

here is the command I ran:
```
from(bucket: "dataset")
  |> range(start: -30d)
  |> filter(fn: (r) => r["_measurement"] == "results")
  |> holtWinters(interval: 1d, n: 10, seasonality: 0, withFit: true)
```

this triggers an error 500 in Grafana
InfluxDB logs revers to the same "internal error not returned to client" as in this issue

**Comment by jithintraderev** [comment]
I also have the same issue with influx 2.6.1. Will be downgrading the versions to check if queries are returned

**Comment by grillbaer** [comment]
Similar problem here with InfluxDB 2.6.1.
After more than 100 successful queries further queries fail.
Client sees socket closed, influxdb logs the warning described here.
Restarting influx fixed the issue for a while.

Query:
```
from(bucket: "signals-complete")
  |> range(start: 0, stop: now())
  |> group(columns: ["_measurement", "_field"])
  |> first()
  |> keep(columns: ["_time"])
  |> min(column: "_time")
```
Metrics when failing: [influxdb-metrics.txt](https://github.com/influxdata/influxdb/files/11002172/influxdb-metrics.txt)

**Comment by grillbaer** [comment]
Update for my comment https://github.com/influxdata/influxdb/issues/23907#issuecomment-1473902321 above:

It turned out that a client-side socket read timeout of 10 seconds lead to the aborted query and the server side warn log message. 
The default for the read timeout in the Java SDK influxdb-client-java is only 10 seconds and any queries taking longer will be aborted. Increasing this timeout fixed the problem.

**Comment by max0x7ba** [comment]
> Update for my comment [#23907 (comment)](https://github.com/influxdata/influxdb/issues/23907#issuecomment-1473902321) above:
> 
> It turned out that a client-side socket read timeout of 10 seconds lead to the aborted query and the server side warn log message. The default for the read timeout in the Java SDK influxdb-client-java is only 10 seconds and any queries taking longer to return results will be aborted. Increasing this timeout fixed the problem.

Increasing the client connection timeout from 30s to 90s fixed the issue for me with influxdb2-2.7.1. 

Thank you.
